test/inductor/test_compiled_optimizers.py

```diff
@@ -82,66 +88,59 @@ class KernelCounts(NamedTuple):
+                if name in [
+                    "test_asgd_capturable_cuda",
+                    "test_asgd_maximize_capturable_cuda",
+                    "test_asgd_weight_decay_capturable_cuda",
+                    "test_asgd_weight_decay_maximize_capturable_cuda",
+                ]:
+                    continue
+
+                kwargs["device"] = device
+                if name in KERNEL_COUNT_OVERRIDES:
+                    kwargs["kernel_count"] = KERNEL_COUNT_OVERRIDES[name]
+                else:
+                    kwargs["kernel_count"] = (
+                        KERNEL_COUNTS[optim_info.optim_cls].multitensor
+                        if kwargs.get("foreach", False) and device == "cuda"
+                        else KERNEL_COUNTS[optim_info.optim_cls].singletensor
                     )
 
-                    for key in kwargs:
-                        if key == "lr":
-                            continue
-                        name += "_" + key
+                if kwargs["kernel_count"] is None:
+                    continue
 
-                    name += f"_{device}"
-
-                    # Eager for-loop impl doesn't support capturable ASGD
-                    if name in [
-                        "test_asgd_capturable_cuda",
-                        "test_asgd_maximize_capturable_cuda",
-                        "test_asgd_weight_decay_capturable_cuda",
-                        "test_asgd_weight_decay_maximize_capturable_cuda",
-                    ]:
-                        continue
-
-                    kwargs["foreach"] = foreach
-                    kwargs["device"] = device
-                    if name in KERNEL_COUNT_OVERRIDES:
-                        kwargs["kernel_count"] = KERNEL_COUNT_OVERRIDES[name]
-                    else:
-                        kwargs["kernel_count"] = (
-                            KERNEL_COUNTS[optim_info.optim_cls].multitensor
-                            if foreach and device == "cuda"
-                            else KERNEL_COUNTS[optim_info.optim_cls].singletensor
-                        )
-
-                    if kwargs["kernel_count"] is None:
-                        continue
-
-                    # Note on tolerances:
-                    # test_adadelta_foreach_rho_weight_decay_cuda
-                    # Mismatched elements: 1 / 100 (1.0%)
-                    # Greatest absolute difference: 2.0936131477355957e-05 at index (2, 7) (up to 2e-05 allowed)
-                    # Greatest relative difference: 8.520411211065948e-05 at index (2, 7) (up to 1e-06 allowed)
-                    if optim_info.optim_cls is Adadelta:
-                        kwargs["rtol"] = 2e-5
-                        kwargs["atol"] = 2e-5
-
-                    compiled_opt_db.append((optim_info.optim_cls, name, kwargs))
+                # fused optimizers are disabled
+                if kwargs.get("fused", False):
+                    kwargs["kernel_count"] = 0
```

Comment:
What's the diff between 0 and None here?

Reply:
None gets skipped, zero gets checked

Reply:
Oh, so None = unsupported/disabled and 0 means we actually generate 0 kernels.

Reply:
I just removed this anyway, we test fused in the parity test